include/highfive/bits/H5Slice_traits.hpp

```diff
@@ -49,10 +49,239 @@ class ElementSet {
+    }
+
+    HyperSlab& operator|=(const RegularHyperSlab& sel) {
+        selects.emplace_back(sel, Op::Or);
+        return *this;
+    }
+
+    HyperSlab operator&(const RegularHyperSlab& sel) const {
+        auto ret = *this;
+        ret &= sel;
+        return ret;
+    }
+
+    HyperSlab& operator&=(const RegularHyperSlab& sel) {
+        selects.emplace_back(sel, Op::And);
+        return *this;
+    }
+
+    HyperSlab operator^(const RegularHyperSlab& sel) const {
+        auto ret = *this;
+        ret ^= sel;
+        return ret;
+    }
+
+    HyperSlab& operator^=(const RegularHyperSlab& sel) {
+        selects.emplace_back(sel, Op::Xor);
+        return *this;
+    }
+
+    HyperSlab& notA(const RegularHyperSlab& sel) {
+        selects.emplace_back(sel, Op::NotA);
+        return *this;
+    }
+
+    HyperSlab& notB(const RegularHyperSlab& sel) {
+        selects.emplace_back(sel, Op::NotB);
+        return *this;
+    }
+
+    bool isRegular(const DataSpace& dataspace) const {
+        if (selects.empty()) {
+            return true;
+        }
+
+        auto selection = apply(dataspace);
+        return isRegularHyperslab(selection);
+    }
+
+    DataSpace apply(const DataSpace& space_) const {
+        auto space = space_.clone();
+        for (const auto& sel : selects) {
+            if (sel.op == Op::None) {
+                H5Sselect_none(space.getId());
+            } else {
+                if (H5Sselect_hyperslab(space.getId(), convert(sel.op),
+                                        sel.offset.empty() ? nullptr : sel.offset.data(),
+                                        sel.stride.empty() ? nullptr : sel.stride.data(),
+                                        sel.count.empty() ? nullptr : sel.count.data(),
+                                        sel.block.empty() ? nullptr : sel.block.data()) <
+                    0) {
+                    HDF5ErrMapper::ToException<DataSpaceException>(
+                        "Unable to select hyperslab");
+                }
+            }
+        }
+        return space;
+    }
+
+  private:
+    enum Op {
+        Noop,
+        Set,
+        Or,
+        And,
+        Xor,
+        NotB,
+        NotA,
+        Append,
+        Prepend,
+        Invalid,
+        None,
+    };
+
+    H5S_seloper_t convert(Op op) const {
+        switch (op) {
+        case Noop:
+            return H5S_SELECT_NOOP;
+        case Set:
+            return H5S_SELECT_SET;
+        case Or:
+            return H5S_SELECT_OR;
+        case And:
+            return H5S_SELECT_AND;
+        case Xor:
+            return H5S_SELECT_XOR;
+        case NotB:
+            return H5S_SELECT_NOTB;
+        case NotA:
+            return H5S_SELECT_NOTA;
+        case Append:
+            return H5S_SELECT_APPEND;
+        case Prepend:
+            return H5S_SELECT_PREPEND;
+        case Invalid:
+            return H5S_SELECT_INVALID;
+        default:
+            return H5S_SELECT_INVALID;
```

Comment:
Seems like here we could throw an exception.

Reply:
Fixed in #542